moto/s3/models.py

```diff
@@ -172,11 +172,11 @@ def complete(self, body):
         count = 0
         for pn, etag in body:
             part = self.parts.get(pn)
-            if part is None or part.etag != etag:
+            part_etag = part.etag.replace('"', '')
+            if part is None or part_etag != etag:
```

Comment:
Add in `etag = etag.replace('"', '')` before the if statement otherwise your doing `'"somemd5"' == 'somemd5'`

Reply:
Think the official spec requires the quotes. I wouldn't strip them to fix the test, the test should expect quotes in the first place.